### DISCUSSION TITLE
RLS with Supabase and Auth0

### DISCUSSION THREAD

**Discussion by logemann** [supabase/supabase#2484] [General]
Having an webapp architecture already in place with auth0 as provider. Is there an easy non-server way to use the access_token from Auth0 and pass it to supabase js client lib? I could enrich the access_token via Auth0 actions to include certain fields supabase expects in the token or what is the best approach to this?

Thanks.

**Comment by logemann** [comment]
trying to get an Idea how JWTs exactly work with Supabase. Getting closer but i dont get why this RLS policy doesnt work when trying to add it, See Screenshot.

<img width="1033" alt="Bildschirmfoto 2021-07-22 um 11 36 35" src="https://user-images.githubusercontent.com/346353/126618810-d50c893d-ed48-4036-91e0-4b856da077c3.png">

**Reply by logemann** [reply]
error on SQL console when trying to create the policy is:
**operator does not exist: uuid = character varying**

**Reply by thomasgrastein** [reply]
Hey @logemann
Not sure about integrating Auth0, but the issue you're facing with the policy seems to be related to comparing uuid with varchar. In your public.user table, the user_id should be of type uuid to properly relate to the uuid of the auth.users table.

**Reply by logemann** [reply]
@thomasgrastein thanks. That was the reason. Another problem that arise is that Auth0 doesnt use UUID in the "sub" :-) And having looked at Hasura JWT handling... They are better in a way that i dont need custom Tokens because i can supply them my JWK (signing key) which i get from Auth0.

With Supabase, the JWK is fixed and forces me to create a custom token based on that signing key. At least thats what it looks like.
